miscellaneous/metaDataRemover/tokenSorting.go

```diff
@@ -0,0 +1,200 @@
+package main
+
+import (
+	"fmt"
+	"math/big"
+	"sort"
+	"strings"
+)
+
+type interval struct {
+	start uint64
+	end   uint64
+}
+
+type tokenWithInterval struct {
+	tokenID  string
+	interval *interval
+}
+
+type tokenData struct {
+	tokenID   string
+	intervals []*interval
+}
+
+func sortTokensIDByNonce(tokens map[string]struct{}) (map[string][]uint64, error) {
```

Comment:
this could have been part of the struct that also held the data. For example, you could have had a data structure `token` like this:
```go
type token struct{
    identifier string
    nonce      uint64
}
```
of course with an `NewToken` constructor function that could have populated both fields

and the map that contains the tokens on a shard could have actually held slice
```go
type tokensOnShard struct{
    tokens  []token
    shardID uint32
}
```

and then, you could have just a sort function call on the slice

Reply:
> type token struct{
    identifier string
    nonce      uint64
}

I honestly disagree with this approach.
I don't see a strong benefit of creating a struct that should do on constructor exactly what this func does.
Why over-complicate things when you have an input and need an output? That's the job of a simple func.

> type tokensOnShard struct{

Also disagree here. Every sorting done in `tokenSorting.go` is independent of any shard, they just sort tokens. Also, why would I create a struct wich has `shardID` as a field, when I feel like is more natural to have a map to sort the tokens by shardID ?